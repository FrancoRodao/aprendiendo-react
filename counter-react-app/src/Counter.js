import React, { useState } from 'react'
import PropTypes from 'prop-types'


const Counter = ({value = 0})=>{
  
  const [counter, setCounter] = useState(value)
  const sumar = () => setCounter(counter+1)

  const restar = () => setCounter(counter-1)

  return (
  <>
    <h1>Counter app</h1>
    <p>valor actual: {counter}</p>
    <button onClick={sumar}>+1</button>
    <button onClick={restar}>-1</button>
  </>)
}

Counter.propTypes = {
  value: PropTypes.number
}

export default Counter;
