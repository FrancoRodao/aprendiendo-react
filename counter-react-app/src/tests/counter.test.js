import React from 'react'
import Counter from "../Counter";
import { shallow } from "enzyme";

describe('counter component', () => {
    const counterValue = 3
    let wrapper = shallow(<Counter value={counterValue}></Counter>);

    //reset counter
    beforeEach(()=>{
        wrapper = shallow(<Counter value={counterValue}></Counter>)
    })
    
    test('should show counter component snapshot', () => {
        const wrapper = shallow(<Counter></Counter>)
        expect(wrapper).toMatchSnapshot()
    })

    test('should show a number sent by props', () => {
        const counterValue = 3
        const textCounter = wrapper.find("p").text()
        expect(textCounter).toBe(`valor actual: ${counterValue}`)
    })

    test('should increment counter on press button +1', () => {
       wrapper.find('button').at(0).simulate("click")
       const textCounter = wrapper.find("p").text()
       expect(textCounter).toBe(`valor actual: ${counterValue+1}`)
    })

    test('should increment counter on press button -1', () => {
        wrapper.find('button').at(1).simulate("click")
        const textCounter = wrapper.find("p").text()
        expect(textCounter).toBe(`valor actual: ${counterValue-1}`)
     })
    
})
