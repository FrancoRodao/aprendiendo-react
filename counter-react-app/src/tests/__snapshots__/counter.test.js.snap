// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter component should show counter component snapshot 1`] = `
<Fragment>
  <h1>
    Counter app
  </h1>
  <p>
    valor actual: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
